Doesn't compile on Adafruit feather m0

1.0.6 compiles fine, but 1.0.7 & 1.0.8 does not
This is compiler error


 In file included from /private/var/folders/b0/tyrz_hq93ln1r0lw4h39dby80000gp/T/.arduinoIDE-unsaved2025313-75751-nmsral.tbpj/Basics_Serial/Basics_Serial.ino:9:
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath.h: In function 'void FixMathPrivate::static_autotests()':
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath_Autotests.h:54:40:   in 'constexpr' expansion of 'SFix<7, 0, 128>(-128, false)'
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath.h:784:101: error: left operand of shift expression '(-128 << 0)' is negative [-fpermissive]
  784 |   constexpr SFix(T value,bool as_raw=false) : internal_value(as_raw ? value : (internal_type(value) << NF)) {};
      |                                                                               ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
In file included from /Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath.h:1401,
                 from /private/var/folders/b0/tyrz_hq93ln1r0lw4h39dby80000gp/T/.arduinoIDE-unsaved2025313-75751-nmsral.tbpj/Basics_Serial/Basics_Serial.ino:9:
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath_Autotests.h:55:35: error: non-constant condition for static assertion
   55 |       static_assert((s+s).getNI() == 8, "test fail");
      |                     ~~~~~~~~~~~~~~^~~~
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath_Autotests.h:56:34: error: non-constant condition for static assertion
   56 |       static_assert((-s).getNI() == 8, "test fail");
      |                     ~~~~~~~~~~~~~^~~~
In file included from /private/var/folders/b0/tyrz_hq93ln1r0lw4h39dby80000gp/T/.arduinoIDE-unsaved2025313-75751-nmsral.tbpj/Basics_Serial/Basics_Serial.ino:9:
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath_Autotests.h:59:43:   in 'constexpr' expansion of 'SFix<1, 0>(-1, false)'
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath.h:784:101: error: left operand of shift expression '(-1 << 0)' is negative [-fpermissive]
  784 |   constexpr SFix(T value,bool as_raw=false) : internal_value(as_raw ? value : (internal_type(value) << NF)) {};
      |                                                                               ~~~~~~~~~~~~~~~~~~~~~~^~~~~~
In file included from /Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath.h:1401,
                 from /private/var/folders/b0/tyrz_hq93ln1r0lw4h39dby80000gp/T/.arduinoIDE-unsaved2025313-75751-nmsral.tbpj/Basics_Serial/Basics_Serial.ino:9:
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath_Autotests.h:60:40: error: non-constant condition for static assertion
   60 |       static_assert((zero - s).getNI() == 8, "test fail");
      |                     ~~~~~~~~~~~~~~~~~~~^~~~
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath_Autotests.h:61:33: error: non-constant condition for static assertion
   61 |       static_assert(-(zero - s) == s, "test fail");
      |                     ~~~~~~~~~~~~^~~~
/Users/bgw-developer/Documents/Arduino/libraries/FixMath/src/FixMath_Autotests.h:62:40: error: non-constant condition for static assertion
   62 |       static_assert((s*negone).getNI() == 8, "test fail");
      |                     ~~~~~~~~~~~~~~~~~~~^~~~
exit status 1

Compilation error: exit status 1

--- a/.github/workflows/compile_examples.yml
+++ b/.github/workflows/compile_examples.yml
@@ -63,6 +63,36 @@ jobs:
               - name: STMicroelectronics:stm32
                 source-url: https://github.com/stm32duino/BoardManagerFiles/raw/main/package_stmicroelectronics_index.json
             internalid: stm32_official
+          - fqbn: esp8266:esp8266:huzzah
+            type: 8266
+            platforms: |
+              - name: esp8266:esp8266
+                source-url: https://arduino.esp8266.com/stable/package_esp8266com_index.json
+            internalid: esp8266
+          - fqbn: arduino:mbed_giga:giga
+            platforms: |
+              - name: arduino:mbed_giga
+            internalid: arduino_giga
+          - fqbn: rp2040:rp2040:rpipico
+            type: rp2040
+            platforms: |
+              - name: rp2040:rp2040
+                source-url: https://github.com/earlephilhower/arduino-pico/releases/download/global/package_rp2040_index.json
+            internalid: rp2040
+          - fqbn: arduino:renesas_uno:minima
+            platforms: |
+              - name: arduino:renesas_uno
+            internalid: arduino_unor4
+          - fqbn: teensy:avr:teensy36
+            platforms: |
+              - name: teensy:avr
+                source-url: https://www.pjrc.com/teensy/package_teensy_index.json
+            internalid: teensy36
+          - fqbn: teensy:avr:teensy41
+            platforms: |
+              - name: teensy:avr
+                source-url: https://www.pjrc.com/teensy/package_teensy_index.json
+            internalid: teensy41
 
     steps:
       - name: Checkout repository

--- a/library.properties
+++ b/library.properties
@@ -1,5 +1,5 @@
 name=FixMath
-version=1.0.8
+version=1.0.9
 author=Thomas Combriat and Thomas Friedrichsmeier
 maintainer=Thomas Combriat <tomcombriat@live.fr>
 sentence=Fixed Point Arithmetics for Arduino and others

--- a/src/FixMath_Autotests.h
+++ b/src/FixMath_Autotests.h
@@ -8,11 +8,11 @@
  *
  */
 
-#if (defined(__GNUC__) && (__GNUC__ < 10)) || (__cplusplus >= 202002L)
+#if (defined(__GNUC__) && (__GNUC__ < 12) && !defined(ESP8266)) && !defined(ARDUINO_ARCH_SAMD) || (__cplusplus >= 202002L)
 // Bit-shifting negative number has not been formally defined behavior before C++ 2020, but rather technically "implementation defined". It thus
 // could not be used in constexpr statements (although working quite fine, at runtime).
 // At the time of this writing (07/2024), we do not have a c++ 2020 compiler running in our automated test workflow. However, older versions of GCC
-// are less pedantic, and will regard at least the case of shift by 0 bits (-1 << 0) as defined. We leverage this fact to compile as many checks as possible.
+// are less pedantic, at least on some platforms (not ESP8266 and SAMD apparently), and will regard at least the case of shift by 0 bits (-1 << 0) as defined. We leverage this fact to compile as many checks as possible.
 #define SHIFT_NEGATIVE_BY_ZERO_DEFINED 1
 #else
 #define SHIFT_NEGATIVE_BY_ZERO_DEFINED 0

